tests/golang/hello/YBuild

```diff
@@ -1,3 +1,12 @@
-GoProg('hello', sources='hello.go', in_buildenv='//:builder')
+GoProg('hello',
+       sources='hello.go',
+       in_buildenv='//:builder',
+       mod_file='go.mod')
+
+GoTest('hello_test',
+       sources='hello_test.go',
+       deps=':hello',
```

Comment:
It's not about testing functions in main module, it's about forcing the Prog to produce the binary artifact when you just want to test - this could be expensive.

Is there a way around it?